test/AutoDiff/SIL/Serialization/differentiable_function_inst.sil

```diff
@@ -0,0 +1,81 @@
+// Round-trip parsing/printing test.
+
+// RUN: %target-sil-opt -enable-experimental-differentiable-programming %s -emit-sorted-sil | %FileCheck %s --check-prefix=CHECK-SIL
+
+// Round-trip serialization-deserialization test.
```

Comment:
Since this is a serialization test, why is it also testing IRGen? Should there be a standalone IRGen test instead?

Reply:
Many SIL tests test multiple functionality on the same code, to avoid code duplication:
- Round-trip parsing and printing.
- Round-trip serialization and deserialization.
- IRGen.

Examples:
- [`test/AutoDiff/SIL/Serialization/sil_differentiability_witness.sil`](https://github.com/apple/swift/blob/master/test/AutoDiff/SIL/Serialization/sil_differentiability_witness.sil)
- [`test/AutoDiff/SIL/Serialization/differentiability_witness_function_inst.sil`](https://github.com/apple/swift/blob/master/test/AutoDiff/SIL/Serialization/differentiability_witness_function_inst.sil)

---

So far, I just shoved these tests randomly in `test/AutoDiff/SIL/Serialization`, which isn't the best location. Suggestions welcome!

Ideas:
- We could move these multi-functionality tests to a parent directory, e.g. `test/AutoDiff/SIL`.
  - I like this option because it's simple.
- We could put multiple copies of these tests in `Parse`/`Serialization`/`IRGen` subdirectories for better organization, but duplicating Swift/SIL code isn't ideal.
- We could keep the Swift/SIL code in an `test/AutoDiff/SIL/Inputs` directory so `Parse`/`Serialization`/`IRGen` tests just have `RUN` lines. But separating code from `RUN` lines isn't very nice.

Reply:
> * We could move these multi-functionality tests to a parent directory, e.g. `test/AutoDiff/SIL`.

Done in https://github.com/apple/swift/pull/30598.